zone/exp.cpp

```diff
@@ -541,6 +542,22 @@ void Client::AddEXP(uint32 in_add_exp, uint8 conlevel, bool resexp) {
 		aaexp = had_aaexp;	//watch for wrap
 	}
 
+	// Check for Unused AA Cap.  If at or above cap, set AAs to cap, set aaexp to 0 and set aa percentage to 0.
+	// Doing this here means potentially one kill wasted worth of experience, but easiest to put it here than to rewrite this function.
+	if (aaexp > 0) {
+		int aa_cap = RuleI(AA, UnusedAAPointCap);
+		if (m_pp.aapoints == aa_cap) {
+			MessageString(Chat::Red, AA_CAP);
+			aaexp = 0;
+			m_epp.perAA = 0;
+		} else if (m_pp.aapoints > aa_cap) {
+			MessageString(Chat::Red, OVER_AA_CAP);
+			m_pp.aapoints = aa_cap;
+			aaexp = 0;
+			m_epp.perAA = 0;
+		}
```

Comment:
If both trees set these 2 variables to 0, can have the setters outside of either condition specifically.

Reply:
Not sure that would work as outside the if/elseif they may be within the correct range.  I could completely flip the logic so its checking if you are over the AA Cap prior to checking if you have aaexp enabled, but I figured that would create needless checks. This way if a player is reimbursed AA's by a GM and had their AAEXP also set to 0 it wouldn't trigger the wipe. Its messy for sure.